src/lib/utils/viewer.ts

```diff
@@ -0,0 +1,39 @@
+import { getContext } from "svelte";
+import type { Document, Note, ViewerMode } from "$lib/api/types";
+import { canonicalUrl, pageHashUrl } from "../api/documents";
+import { noteHashUrl } from "../api/notes";
+
+interface ViewerHrefOptions {
+  document?: Document;
+  page?: number;
+  note?: Note;
+  mode?: ViewerMode;
+  embed?: boolean;
+}
+
+export function isEmbedded(): boolean {
+  // are we embedded?
+  return getContext("embed") ?? false;
+}
+
+export function getViewerHref(options: ViewerHrefOptions = {}) {
```

Comment:
We should have some tests for this function, since it has so many options and is used all over.